rust/ql/lib/codeql/rust/controlflow/internal/CfgNodes.qll

```diff
@@ -0,0 +1,148 @@
+private import rust
+private import codeql.rust.controlflow.internal.generated.CfgNodes
+private import codeql.rust.controlflow.internal.ControlFlowGraphImpl as CfgImpl
+private import codeql.rust.controlflow.ControlFlowGraph
+private import codeql.rust.controlflow.BasicBlocks
+private import codeql.rust.controlflow.CfgNodes
+private import codeql.rust.internal.CachedStages
+
+private predicate isPostOrder(AstNode n) {
+  n instanceof Expr and
+  not n instanceof LetExpr
```

Comment:
Maybe it doesn't matter, but `ParenExpr` are also not in post order.
```suggestion
  not n instanceof LetExpr and
  not n instanceof ParenExpr and
```

Reply:
`ParenExpr`s are not part of the CFG, instead they are considered to be desugared.